cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -102,7 +103,13 @@ func buildKubeletArgMapCommon(opts kubeletFlagsOpts) map[string]string {
 	// Once that happens only the "remote" branch option should be left.
 	// TODO: https://github.com/kubernetes/kubeadm/issues/2626
 	hasDockershim := opts.kubeletVersion.Major() == 1 && opts.kubeletVersion.Minor() < 24
-	if opts.nodeRegOpts.CRISocket == constants.DefaultDockerCRISocket && hasDockershim {
+	var dockerSocket string
```

Comment:
Maybe we can just rename `DefaultDockerCRISocket` to `CRISocketDockershim` in `constants_unix.go/constants_windows.go` and mark it as deprecated and only for compatibility?

Reply:
this commit changes DefaultCRISocket to equal the containerd socket and removes constants_*.go exposure of the dockershim socket with the idea that kubeadm would not know about it after 1.25 and it will be unsupported.

we could keep CRISocketDockershim in constants_*.go for one more release marked as deprecated, but not sure who would use it. for example, i know that minikube forks the kubeadm constants files, but i don't think they use the dockershim socket.

we currently consider the kubeadm files as not importable and we have not support guarantees around publicly exposed constants/functions in kubeadm outside of the API packages.

with that in mind i inlined the socket strings in this function, but if you think it is better to have CRISocketDockershim for one more release i can add it with the same TODO on top of it:
```
// TODO: https://github.com/kubernetes/kubeadm/issues/2626
```

Reply:
OK, if this is a deliberate attempt to avoid misuse, that is fine. And we can clean this logic in a future release.

Reply:
minikube uses `kubeadm` (and `kubectl`) from the kubernetes version to be installed, there is no version skew.

I thought the decision was to **not** have any default, and have CRI "crash" on start the same way that CNI does ?

Reply:
> I thought the decision was to not have any default, and have CRI "crash" on start the same way that CNI does ?

The default would be visible if the user generates a config file with the 'kubeadm config print*' commands. We call that a static default and it is also used across the codebase for unit tests. It acts like a placeholder / example value.

But on runtime we have the dynamic detection that will try to populate the field (if empty) and error out on multiple or 0 sockets found on the host.